src/responder/pam/pamsrv_gssapi.c

```diff
@@ -385,12 +469,130 @@ static char *gssapi_get_name(TALLOC_CTX *mem_ctx, gss_name_t gss_name)
+
+    major = gss_inquire_name(&minor, gss_name, &is_mechname, NULL, &attrs);
+    if (major != GSS_S_COMPLETE) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Unable to inquire name\n");
+        return NULL;
+    }
+
+    if (attrs == GSS_C_NO_BUFFER_SET) {
+        DEBUG(SSSDBG_TRACE_FUNC, "No krb5 attributes in the ticket\n");
+        return NULL;
+    }
+
+    exported = talloc_strdup(mem_ctx, "");
+    if (exported == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Unable to pre-allocate indicators\n");
+        return NULL;
+    }
+
+    for (int i = 0; i < attrs->count; i++) {
+        gss_buffer_desc value;
+        gss_buffer_desc display_value;
+        int authenticated = 0;
+        int complete = 0;
+        int more = -1;
+
+        /* skip anything but auth-indicators */
+        if (strncmp(AUTH_INDICATORS_TAG, attrs->elements[i].value,
+                    sizeof(AUTH_INDICATORS_TAG) - 1) != 0)
+            continue;
+
+        /* retrieve all indicators */
+        while (more != 0) {
+            value.value = NULL;
+            display_value.value = NULL;
+
+            major = gss_get_name_attribute(&minor, gss_name,
+                                            &attrs->elements[i],
+                                            &authenticated,
+                                            &complete, &value,
+                                            &display_value,
+                                            &more);
+            if (major != GSS_S_COMPLETE) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                        "Unable to retrieve an attribute\n");
+                gss_release_buffer(&minor, &value);
+                gss_release_buffer(&minor, &display_value);
+                TALLOC_FREE(exported);
+                (void)gss_release_buffer_set(&minor, &attrs);
+                return NULL;
+            }
+
+            if ((value.value != NULL) && authenticated) {
+                DEBUG(SSSDBG_TRACE_FUNC,
+                        "attribute's [%.*s] value [%.*s] authenticated\n",
+                        (int) attrs->elements[i].length,
+                        (char*) attrs->elements[i].value,
+                        (int) value.length,
+                        (char*) value.value);
+                exported = talloc_asprintf_append(exported, "%.*s ",
+                                                (int) value.length,
+                                                (char*) value.value);
+            }
+
+            if (exported == NULL) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                        "Unable to collect an attribute value\n");
+                (void)gss_release_buffer(&minor, &value);
+                (void)gss_release_buffer(&minor, &display_value);
+                (void)gss_release_buffer_set(&minor, &attrs);
+                return NULL;
+            }
+            (void)gss_release_buffer(&minor, &value);
+            (void)gss_release_buffer(&minor, &display_value);
```

Comment:
While I think it would be better to use `goto done` cleanup for this function as a whole, if you choose not to do that, these two lines can go before the `if` above.